test/metabase/api/public_test.clj

```diff
@@ -144,6 +146,20 @@
             (is (= "Not found."
                    (client/client :get 404 (str "public/card/" uuid))))))))))
 
+(deftest public-queries-are-counted-test
+  (testing "GET /api/public/card/:uuid/query coutns as a public query"
+    (mt/with-temporary-setting-values [enable-public-sharing true]
+      (with-temp-public-card [{uuid :public_uuid}]
+        (testing "should increment the public link query count when fetching a public Card"
+          (let [get-qe-count (fn get-qe-count [] (get-in (#'stats/->snowplow-grouped-metric-info)
+                                                         [:query-executions "public_link"]))
+                qe-count-before (get-qe-count)]
+            (client/client :get 202 (str "public/card/" uuid "/query"))
+            ;; The count gets incremented asynchronously, so we need to poll until it's updated:
+            (testing "the count should be incremented within 300 ms:"
+              (is (true?
```

Comment:
nit: are we worried about this returning truthiness instead of `true`? If not just `(is (tu/poll-until ...))` seems fine, but seems fine either way

Reply:
good point, I'll update it.